src/coreComponents/mesh/ElementRegionManager.cpp

```diff
@@ -152,6 +154,78 @@ void ElementRegionManager::generateMesh( CellBlockManagerABC const & cellBlockMa
   } );
 
 }
+void ElementRegionManager::checkSubRegionRegistering( CellBlockManagerABC const & cellBlockManager )
+{
+  Group const & cellBlocks = cellBlockManager.getCellBlocks();
+  //std::map< string, CellElementSubRegion const * > cellBlocksSubRegion;
+
+  // region that contain a cellBlock (name)
+  std::map< string, CellElementRegion const * > cellBlocksRegion;
+
+  // Let's find out which CellElementSubRegion is using each source cellBlock.
+  forElementRegions< CellElementRegion >( [&]( CellElementRegion & region ) {
+    region.forElementSubRegions< CellElementSubRegion >( [&]( CellElementSubRegion & subRegion ) {
+      string const & subRegionName = subRegion.getName();
+
+      // if no region already referenced the cellBlock
+      if( cellBlocksRegion.find( subRegionName ) == cellBlocksRegion.end() )
```

Comment:
Use `std::map::insert` instead?

Reply:
Would `emplace` do the job?
```c++
      string const & subRegionName = subRegion.getName();
      const auto [regionIterator, isNewRegion] = cellBlocksRegion.emplace( subRegionName, &region );
      // if no region already referenced the cellBlock
      if( !isNewRegion )
      {
        CellElementRegion const & otherRegion = iterator->second;
```

Reply:
Yes, it should be equivalent. The idea is that you'll search only once (and if you insert twice, then you exhibit an error so we don't car).